Modify error message when adding a collaborator.  Closes #1331

parrish opened the pull request.

chrissnyder commented on app/pages/lab/collaborators.cjsx:
```diff
@@ -119,7 +119,12 @@ CollaboratorCreator = React.createClass
+          error.message = 'Your account status on this project is still being setup. Please try again later.'
+
+        creating = false
+        @setState {error, creating}
```
Shouldn't need to set `creating` as the following then block will run.

chrissnyder commented on app/pages/lab/collaborators.cjsx:
```diff
@@ -119,7 +119,12 @@ CollaboratorCreator = React.createClass
+
+        creating = false
+        @setState {error, creating}
+        @fetchAllRoles()
```
Doesn't exist in this component. Fire a callback instead?

chrissnyder commented:
Any trick to testing this/simulating heavier API load? Look fine in abstract, just curious if I'm missing something.

parrish commented:
Updated.  To simulate the condition of having the panoptes role, but not the talk role, I just removed the talk role manually.

<img width="1114" alt="screen shot 2015-10-12 at 2 05 41 pm" src="https://cloud.githubusercontent.com/assets/122990/10436198/8abe5c5e-70ea-11e5-932c-482d6ed39b1f.png">

chrissnyder commented:
:+1:
